More parallel test fixes

roystgnr opened the pull request:
A couple of the failing-at-higher-processor-count tests fixed here were just silly FP error or linear solver problems, but investigating these test failures also uncovered #9576, which I believe should be completely fixed here.

roystgnr committed:
5e-9 floor on eigen inverse_power test

We hit a 2.1e-9 relative error on 11 processors with Civet, which is
surely a false positive.

roystgnr committed:
Don't try computing an AuxScalarKernel if we can't

If the kernel is for a SCALAR whose support doesn't include a
particular processor, then trying to compute on that processor will be
an assertion failure in dbg mode and undefined behavior in opt.

This fixes the correctness issues in #9576 for me, including what was
previously a failure on 13+ processors in #9430

roystgnr committed:
Avoid redundant MooseVariableScalar::insert()

This should fix the parallel scaling issue discussed in #9576.

This is a pretty deep change, so in theory there might be side
effects... but I can't picture anything being broken after this change
unless it was already broken-with-a-race-condition beforehand, e.g. if
different processors computed different values for the same scalar
variable.

roystgnr committed:
Limit 1d_dg_block_restrict test to 12 procs

I verified that it works fine with MUMPS LU, so the convergence
failures we see at around 15 processors are just preconditioning
or iterative linear solver issues.  However, I don't see any *other*
MOOSE tests using MUMPS, and I don't want to be the one to introduce
that dependency, and neither of my other go-to preconditioner options
(BoomerAMG, ASM+ILU levels cranked up) worked readily, so we'll just
leave this as a low-processor-count test for now.

moosebuild commented:
Job [Precheck](https://www.moosebuild.org/job/104057/) on f888ddf wanted to post the following:

Your code requires style changes.
In the top level of your repository, you can run:

`git clang-format b6ecd143451c70db01f7a6fc697bf35af7738fac`

Alternatively, apply the patch found [here](http://mooseframework.org/docs/PRs/9581/style.patch).
You can directly apply the patch by running:

`curl -s http://mooseframework.org/docs/PRs/9581/style.patch | git apply -v`

roystgnr committed:
clang-format changes

"Lock up your libraries if you like; but there is no gate, no lock, no
bolt that you can set upon the freedom of my mind." - Virginia Woolf

moosebuild commented:
Job [Documentation](https://www.moosebuild.org/job/104103/) on 9ddf361 wanted to post the following:

View the site [here](http://mooseframework.org/docs/PRs/9581/site)

roystgnr commented:
Can someone turn on the moosebuild Parallel Test before this gets merged?  I'd like to see if Civet's experiences agree with mine.

roystgnr commented:
Yeah, that's exactly what I get.  Success through -p 12, ```postprocessors/scalar_coupled_postprocessor.test``` fails at 15 (where Metis inexplicably starves processor 0 of elements), and ```userobjects/shape_element_user_object.shape_element_user_object``` fails at 13 and up (also due to a combination of Metis sucking and MOOSE code failing to handle the "processor with nothing to do but twiddle its thumbs" corner case, but with a little "libMesh code failing to distribute nodes equitably" thrown in).

Anyway, everything I fixed seems fixed, so this should be ready to merge.
